login/service_blackbox_test.go

```diff
@@ -0,0 +1,198 @@
+package login_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app"
+	. "github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+)
+
+var db *gorm.DB
+var loginService Service
+
+func TestMain(m *testing.M) {
+	if _, c := os.LookupEnv(resource.Database); c == false {
+		fmt.Printf(resource.StSkipReasonNotSet+"\n", resource.Database)
+		return
+	}
+
+	dbhost := os.Getenv("ALMIGHTY_DB_HOST")
+	if "" == dbhost {
+		panic("The environment variable ALMIGHTY_DB_HOST is not specified or empty.")
+	}
+	var err error
+	db, err = gorm.Open("postgres", fmt.Sprintf("host=%s database=postgres user=postgres password=mysecretpassword sslmode=disable", dbhost))
+	if err != nil {
+		panic("failed to connect database: " + err.Error())
+	}
+	defer db.Close()
+
+	oauth := &oauth2.Config{
+		ClientID:     "875da0d2113ba0a6951d",
```

Comment:
This should come from the configuration as well. This might not be available yet, but..